<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yang &amp; Steve" />


<title>5_WGCNA</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">20190717_Lardelli_RNASeq_Larvae</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">5_WGCNA</h1>
<h4 class="author">Yang &amp; Steve</h4>
<h4 class="date">29/09/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-10-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>20190717_Lardelli_RNASeq_Larvae/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200227code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200227)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200227code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200227)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
Soft Thresholding Power Plots
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>5_WGCNA_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyangdongau20190717LardelliRNASeqLarvaetree417413ae5622d20bb01023802f38d7ba34c415c1targetblank417413aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/yangdongau/20190717_Lardelli_RNASeq_Larvae/tree/417413ae5622d20bb01023802f38d7ba34c415c1" target="_blank">417413a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomyangdongau20190717LardelliRNASeqLarvaetree417413ae5622d20bb01023802f38d7ba34c415c1targetblank417413aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    1_DE-gene-analysis_cache/
    Ignored:    1_DE-gene-analysis_files/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/.Rproj.user/
    Ignored:    analysis/5_WGCNA_cache/
    Ignored:    data/.DS_Store
    Ignored:    data/0_rawData/.DS_Store
    Ignored:    data/1_trimmedData/.DS_Store
    Ignored:    data/2_alignedData/.DS_Store
    Ignored:    files/
    Ignored:    keggdiagram/.DS_Store
    Ignored:    output/.DS_Store

Unstaged changes:
    Modified:   analysis/index.Rmd

Staged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yangdongau/20190717_Lardelli_RNASeq_Larvae/blob/417413ae5622d20bb01023802f38d7ba34c415c1/analysis/5_WGCNA.Rmd" target="_blank">417413a</a>
</td>
<td>
yangdongau
</td>
<td>
2020-10-28
</td>
<td>
Add in WGCNA and variance partitioning analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yangdongau/20190717_Lardelli_RNASeq_Larvae/417413ae5622d20bb01023802f38d7ba34c415c1/docs/5_WGCNA.html" target="_blank">417413a</a>
</td>
<td>
yangdongau
</td>
<td>
2020-10-28
</td>
<td>
Add in WGCNA and variance partitioning analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level3">
<h3>Setup</h3>
<pre class="r"><code>library(dplyr)
library(magrittr)
library(reshape2)
library(readr)
library(tibble)
library(tidyverse)
library(pander)
library(biomaRt)
library(limma)
library(edgeR)
library(genefilter)
library(WGCNA)
library(dynamicTreeCut)
library(ggplot2)
library(RColorBrewer)
library(grid)
  # Viewpoints for showing 4 plots on a 2x2 grid
  vp1 &lt;- viewport(x = 0, y = 0, width = 0.5, height = 0.5, just = c(0, 0))
  vp2 &lt;- viewport(x = 0.5, y = 0, width = 0.5, height = 0.5, just = c(0,0))
  vp3 &lt;- viewport(x = 0, y = 0.5, width = 0.5, height = 0.5, just = c(0, 0))
  vp4 &lt;- viewport(x = 0.5, y = 0.5, width = 0.5, height = 0.5, just = c(0, 0))
library(kableExtra)
library(pheatmap)
library(anRichment)

theme_set(theme_bw())
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;table.style&quot;, &quot;rmarkdown&quot;)
if (interactive()) setwd(here::here(&quot;analysis&quot;))</code></pre>
</div>
<div id="data-load" class="section level1">
<h1>Data load</h1>
<pre class="r"><code>dgeList &lt;- read_rds(here::here(&quot;data&quot;,&quot;dgeList.rds&quot;))
entrezGenes &lt;- dgeList$genes %&gt;%
  dplyr::filter(!is.na(entrez_gene)) %&gt;%
  unnest(entrez_gene) %&gt;%
  dplyr::rename(entrez_gene = entrez_gene)
topTable &lt;- file.path(here::here(&quot;output&quot;, &quot;topTable.csv&quot;)) %&gt;% 
  read_csv()</code></pre>
</div>
<div id="filtering" class="section level1">
<h1>Filtering</h1>
<p>Only keep the genes of high expression (cpm &gt; 1.5 in 6 samples) and high variation (not in the bottom 25%).</p>
<pre class="r"><code>A &lt;- dgeList %&gt;% 
  cpm(log = TRUE) %&gt;% 
  melt %&gt;% 
  dplyr::filter(is.finite(value)) %&gt;% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle(&quot;Before filtering&quot;) +
  labs(x = &quot;logCPM&quot;, y = &quot;Density&quot;)

# Retain genes which are expressed at sufficiently high levels, at least 1.5 cpm in 6 (half of the total number of) samples.
keepTheseGenes &lt;- rowSums(cpm(dgeList) &gt; 1.5) &gt;= 6  

# Density plot of the gene expression distribution after filtering out low expressed genes. 
B &lt;- dgeList %&gt;% 
  cpm(log = TRUE) %&gt;% 
  .[keepTheseGenes,] %&gt;%
  melt %&gt;% 
  dplyr::filter(is.finite(value)) %&gt;% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle(&quot;After filtering low exp genes&quot;)

# Retain genes with sufficiently high variation in expression across all samples.
# Here we define sufficiently high variation as having a standard deviation above the 25th quantile (so not in bottom 25%).
variableGenes &lt;- rowSds(cpm(dgeList, log=TRUE)) &gt;  quantile(rowSds(cpm(dgeList, log=TRUE)), 0.75)

# Density plot of gene expression distribution after filtering out low variance genes. 
C &lt;- dgeList %&gt;% 
  cpm(log = TRUE) %&gt;% 
  .[variableGenes,] %&gt;%
  melt %&gt;% 
  dplyr::filter(is.finite(value)) %&gt;% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle(&quot;After filtering variable genes&quot;)

# Density plot of gene expression distribution after filtering out both low expressed and low variance genes. 
D &lt;- dgeList %&gt;% 
  cpm(log = TRUE) %&gt;% 
  .[keepTheseGenes&amp;variableGenes,] %&gt;%
  melt %&gt;% 
  dplyr::filter(is.finite(value)) %&gt;% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle(&quot;After filtering both&quot;)

# Plot the effect of filtering low expressed genes, low variation genes, and both.
grid.newpage()
print(A, vp = vp3)
print(B, vp = vp4)
print(C, vp = vp1)
print(D, vp  = vp2)</code></pre>
<p><img src="figure/5_WGCNA.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yangdongau/20190717_Lardelli_RNASeq_Larvae/blob/417413ae5622d20bb01023802f38d7ba34c415c1/docs/figure/5_WGCNA.Rmd/unnamed-chunk-1-1.png" target="_blank">417413a</a>
</td>
<td>
yangdongau
</td>
<td>
2020-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dgeList_filtered &lt;- dgeList[keepTheseGenes&amp;variableGenes,,keep.lib.sizes = FALSE]</code></pre>
<p>There are 19396 in dgeList. After filtering, 4144 genes are retained.</p>
</div>
<div id="convert-gene-counts-into-logcpm-form-and-stored-in-gene-count-matraix-x." class="section level1">
<h1>Convert gene counts into logCPM form and stored in gene count matraix <code>X</code>.</h1>
<pre class="r"><code>X &lt;- dgeList_filtered$counts %&gt;% 
  cpm(log=TRUE) %&gt;% t %&gt;% as.data.frame</code></pre>
<div id="check-sample-clustering" class="section level2">
<h2>Check Sample Clustering</h2>
<p>Here we use the gene expression patterns described in <code>X</code> to build sample dendrogram (sample tree), and check how the <code>traits</code> related to the sample dendrogram.</p>
<p><code>traits</code> uses the design matrx used in previous DE genes analysis.</p>
<pre class="r"><code># Apply hierarchial clustering to X. 
sampleTree &lt;- hclust(dist(X), method = &quot;average&quot;)

# Using the design matrix, which contains information of samples&#39; pair and genotype
traits &lt;- expDesign &lt;- model.matrix(~0 + pair + Genotype, dgeList$samples) %&gt;%
  set_colnames(str_replace(colnames(.), pattern = &quot;Genotype.+&quot;, &quot;Mutant&quot;)) %&gt;% as.data.frame

# Assign each group in design matrix to a colour. 
traitColors &lt;- numbers2colors(traits, signed = FALSE, colors = c(&quot;#ffffff&quot;, &quot;#73626E&quot;))

# Plot dendrogram of clustered samples with the design matrix underneath. 
plotDendroAndColors(sampleTree, traitColors,
                    groupLabels = names(traits),
                    main = &quot;Sample dendrogram and trait heatmap&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/5_WGCNA.Rmd/Check%20Sample%20Clustering-1.png" alt="Hierarchial clustering of Q96K97del larvae RNA-seq libraries based on expression of 13,538 genes in the matrix X. " width="672" />
<p class="caption">
Hierarchial clustering of Q96K97del larvae RNA-seq libraries based on expression of 13,538 genes in the matrix X.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Check-Sample-Clustering-1">
Past versions of “Check Sample Clustering-1.png”
</button>
</p>
<div id="fig-no-spaces-Check-Sample-Clustering-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yangdongau/20190717_Lardelli_RNASeq_Larvae/blob/417413ae5622d20bb01023802f38d7ba34c415c1/docs/figure/5_WGCNA.Rmd/Check Sample Clustering-1.png" target="_blank">417413a</a>
</td>
<td>
yangdongau
</td>
<td>
2020-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="choosing-the-soft-thresholding-power" class="section level3">
<h3>Choosing the soft-Thresholding Power</h3>
<p>Constructing a weighted gene network requires a soft-thresholding power β, which will be chosen based on the scale-free topology criterion described by <a href="https://doi.org/10.2202/1544-6115.1128">Zhang &amp; Horvath (2005)</a>.</p>
<p>Here we performs the analysis of network topology for choosing a proper soft-thresholding power.</p>
<p>The following plots show the sufficiently high scale-free topology index (R<sub>2</sub>). We choose 13 as soft thresholding power, which is the lowest power reaching a high value (roughly 0.8).</p>
<pre class="r"><code># Choose a set of soft thresholding powers and call the network topology analysis fucntion.
# Here I use a signed network 
powers &lt;- c(c(1:10), seq(from = 12, to = 26, by = 2))
sft &lt;- pickSoftThreshold(X, powerVector = powers, verbose = 5, networkType = &quot;signed&quot;)</code></pre>
<pre><code>pickSoftThreshold: will use block size 4144.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 4144 of 4144
   Power SFT.R.sq   slope truncated.R.sq mean.k. median.k. max.k.
1      1 1.50e-01  4.7600          0.663  2090.0    2090.0   2240
2      2 2.92e-05 -0.0383          0.472  1230.0    1200.0   1510
3      3 5.93e-02 -0.8130          0.669   799.0     778.0   1130
4      4 1.16e-01 -0.7250          0.794   557.0     537.0    899
5      5 2.18e-01 -0.7320          0.848   408.0     388.0    741
6      6 3.50e-01 -0.7880          0.875   311.0     290.0    626
7      7 4.84e-01 -0.8500          0.893   245.0     222.0    541
8      8 5.96e-01 -0.9330          0.891   197.0     174.0    475
9      9 6.85e-01 -0.9820          0.899   162.0     138.0    422
10    10 7.39e-01 -1.0100          0.904   136.0     111.0    378
11    12 7.76e-01 -1.0600          0.895    98.7      74.5    311
12    14 8.17e-01 -1.0900          0.895    74.9      52.6    262
13    16 8.41e-01 -1.1000          0.900    58.7      38.4    225
14    18 8.69e-01 -1.1100          0.915    47.1      28.5    196
15    20 8.77e-01 -1.1400          0.924    38.6      21.6    175
16    22 8.56e-01 -1.1800          0.911    32.2      16.5    158
17    24 8.56e-01 -1.2100          0.913    27.2      12.9    144
18    26 8.46e-01 -1.2400          0.908    23.2      10.2    132</code></pre>
<pre class="r"><code># Plot the effect of soft-thresholding power on network topologies. 
sizeGrWindow(9, 5)
par(mfrow = c(1,2))
cex1 = 0.9
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab=&quot;Soft Threshold (power)&quot;,ylab=&quot;Scale Free Topology Model Fit,signed R^2&quot;,type=&quot;n&quot;,
     main = paste(&quot;Scale independence&quot;));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2], labels=powers,cex=cex1,col=&quot;red&quot;)
# this line corresponds to using an R^2 cut-off of 0.80.
abline(h=0.8,col=&quot;red&quot;)

plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab=&quot;Soft Threshold (power)&quot;,ylab=&quot;Mean Connectivity&quot;, type=&quot;n&quot;,
     main = paste(&quot;Mean connectivity&quot;))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col=&quot;red&quot;)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Choose the power 13.</p>
</div>
</div>
<div id="step-by-step-network-construction" class="section level2">
<h2>Step by step network construction</h2>
<pre class="r"><code>softPower &lt;- 13

# Calculate the adjacency matrix, using the soft thresholding power 13
adjacency_zebrafish &lt;- adjacency(X, power = softPower, type = &quot;signed&quot;)

# To minimise effects of noise and spurious correlations, we transform the adjacency into a Topological Overlap Matrix (TOM) and calculate the dissimilarity.
TOM_zebrafish &lt;- TOMsimilarity(adjacency_zebrafish, TOMType = &quot;signed&quot;)</code></pre>
<pre><code>..connectivity..
..matrix multiplication (system BLAS)..
..normalization..
..done.</code></pre>
<pre class="r"><code>dissTOM_zebrafish &lt;- 1-TOM_zebrafish

# We now use hierarchial clustering to produce a hierarchial clustering tree (dendrogram) of genes using TOM
geneTree_zebrafish &lt;- hclust(as.dist(dissTOM_zebrafish), method = &quot;average&quot;)

# Plot the dendrogram
plot(geneTree_zebrafish, xlab = &quot;&quot;, ylab = &quot;&quot;, main = &quot;Gene clustering on TOM-based dissimilarity&quot;, labels = FALSE, hang = 0.04)</code></pre>
<p><img src="figure/5_WGCNA.Rmd/adjacency,%20TOM,%20and%20Clustering%20using%20TOM-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-adjacency,-TOM,-and-Clustering-using-TOM-1">
Past versions of “adjacency, TOM, and Clustering using TOM-1.png”
</button>
</p>
<div id="fig-no-spaces-adjacency,-TOM,-and-Clustering-using-TOM-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yangdongau/20190717_Lardelli_RNASeq_Larvae/blob/417413ae5622d20bb01023802f38d7ba34c415c1/docs/figure/5_WGCNA.Rmd/adjacency, TOM, and Clustering using TOM-1.png" target="_blank">417413a</a>
</td>
<td>
yangdongau
</td>
<td>
2020-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="check-similarity-of-modules-before-merging" class="section level2">
<h2>Check similarity of modules before merging</h2>
<pre class="r"><code>dynamicMods = cutreeDynamic(dendro = geneTree_zebrafish, distM = dissTOM_zebrafish,
              deepSplit = 2, pamRespectsDendro = FALSE,
              minClusterSize = 50)</code></pre>
<pre><code> ..cutHeight not given, setting it to 0.991  ===&gt;  99% of the (truncated) height range in dendro.
 ..done.</code></pre>
<pre class="r"><code>dynamicColors = labels2colors(dynamicMods)
table(dynamicColors)</code></pre>
<pre><code>dynamicColors
       black         blue        brown         cyan        green 
         234          541          501           88          419 
 greenyellow      magenta midnightblue         pink       purple 
         135          194           66          196          154 
         red       salmon          tan    turquoise       yellow 
         393           89          129          558          447 </code></pre>
<pre class="r"><code>MEList = moduleEigengenes(X, colors = dynamicColors)
MEs = MEList$eigengenes
# Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs);
# Cluster module eigengenes
METree = hclust(as.dist(MEDiss), method = &quot;average&quot;);
# Plot the result
sizeGrWindow(7, 6)
plot(METree, main = &quot;Clustering of module eigengenes&quot;,
xlab = &quot;&quot;, sub = &quot;&quot;)</code></pre>
</div>
<div id="final-modules" class="section level2">
<h2>Final modules</h2>
<pre class="r"><code># Using 80 as minimum module size With PAM
treeCut_zebrafish &lt;- dynamicTreeCut::cutreeDynamic(
  dendro = geneTree_zebrafish,
  distM = dissTOM_zebrafish,
  method = &quot;hybrid&quot;,
  minClusterSize = 80,
  minGap = 0.25,
  maxCoreScatter = 0.75,
  verbose = 0,
  pamStage = TRUE,
  #deepSplit = 3,
  cutHeight = 0.8
)

colors_zebrafish &lt;- labels2colors(treeCut_zebrafish)
table(colors_zebrafish)</code></pre>
<pre><code>colors_zebrafish
     blue     brown     green      grey       red turquoise    yellow 
      330       316       161      2586       151       417       183 </code></pre>
<pre class="r"><code>plotDendroAndColors(geneTree_zebrafish, 
                    colors_zebrafish, &quot;Dynamic Tree Cut&quot;,
                    dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05,
                    main = &quot;Gene dendrogram and module colours&quot;)</code></pre>
<p><img src="figure/5_WGCNA.Rmd/Optimised%20Initial%20Dendrogram-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Optimised-Initial-Dendrogram-1">
Past versions of “Optimised Initial Dendrogram-1.png”
</button>
</p>
<div id="fig-no-spaces-Optimised-Initial-Dendrogram-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yangdongau/20190717_Lardelli_RNASeq_Larvae/blob/417413ae5622d20bb01023802f38d7ba34c415c1/docs/figure/5_WGCNA.Rmd/Optimised Initial Dendrogram-1.png" target="_blank">417413a</a>
</td>
<td>
yangdongau
</td>
<td>
2020-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="relating-modules-to-traits" class="section level3">
<h3>Relating modules to <code>traits</code></h3>
<pre class="r"><code># Define numbers of genes and samples
nGenes &lt;- ncol(X)
nSamples &lt;- nrow(X)

# Recalculate the MEs with colour labels
MEs0 &lt;- moduleEigengenes(X, colors_zebrafish)$eigengenes
MEs &lt;- orderMEs(MEs0)


moduleTraitCor &lt;- cor(MEs, traits, use = &quot;p&quot;, method = &quot;spearman&quot;)
# Remove the grey module
moduleTraitCor &lt;- moduleTraitCor[rownames(moduleTraitCor) != &quot;MEgrey&quot;,]
#moduleTraitCor &lt;- bicor(MEs, traits, use = &quot;p&quot;, maxPOutliers = 0.1, robustY=FALSE)
moduleTraitPValue &lt;- corPvalueStudent(moduleTraitCor, nSamples)</code></pre>
</div>
<div id="plot-the-relationships-between-module-and-trait" class="section level3">
<h3>Plot the relationships between module and <code>trait</code></h3>
<pre class="r"><code># Visualise relationships between modules and traits.
# Will display correlations and their p-values
textMatrix &lt;- paste(signif(moduleTraitCor, 2), &quot;\n(&quot;, signif(moduleTraitPValue, 1), &quot;)&quot;, sep = &quot;&quot;)
dim(textMatrix) &lt;- dim(moduleTraitCor)

pheatmap::pheatmap(
  moduleTraitCor,
  color = colorRampPalette(c(&quot;#10c1e5&quot;, &quot;#82e0b4&quot;,&quot;#F9F9F9&quot;, &quot;#FBB829&quot;, &quot;#FF0066&quot;))(100), 
  labels_row = gsub(x = names(MEs), pattern = &quot;ME&quot;, replacement = &quot;&quot;),
  border_color = &quot;white&quot;, 
  treeheight_row = 0, treeheight_col = 0,
  clustering_distance_rows = &quot;correlation&quot;,
  clustering_distance_cols = &quot;correlation&quot;, 
  cutree_cols = 3, 
  cluster_cols = FALSE, number_color = &quot;black&quot;,
  cellwidth = 25, cellheight = 25,fontsize = 9,
  display_numbers = textMatrix 
  )</code></pre>
<p><img src="figure/5_WGCNA.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yangdongau/20190717_Lardelli_RNASeq_Larvae/blob/417413ae5622d20bb01023802f38d7ba34c415c1/docs/figure/5_WGCNA.Rmd/unnamed-chunk-3-1.png" target="_blank">417413a</a>
</td>
<td>
yangdongau
</td>
<td>
2020-10-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>genesInModule &lt;- data.frame(module = colors_zebrafish, gene = colnames(X))

colors_zebrafish %&gt;% table %&gt;% as.data.frame %&gt;% set_colnames(c(&quot;Module&quot;, &quot;Genes&quot;)) %&gt;% dplyr::arrange(desc(Genes)) %&gt;% kable(format=&quot;html&quot;)</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
Module
</th>
<th style="text-align:right;">
Genes
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
grey
</td>
<td style="text-align:right;">
2586
</td>
</tr>
<tr>
<td style="text-align:left;">
turquoise
</td>
<td style="text-align:right;">
417
</td>
</tr>
<tr>
<td style="text-align:left;">
blue
</td>
<td style="text-align:right;">
330
</td>
</tr>
<tr>
<td style="text-align:left;">
brown
</td>
<td style="text-align:right;">
316
</td>
</tr>
<tr>
<td style="text-align:left;">
yellow
</td>
<td style="text-align:right;">
183
</td>
</tr>
<tr>
<td style="text-align:left;">
green
</td>
<td style="text-align:right;">
161
</td>
</tr>
<tr>
<td style="text-align:left;">
red
</td>
<td style="text-align:right;">
151
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="genes-in-each-module" class="section level2">
<h2>genes in each module</h2>
<pre class="r"><code>genesInModule %&gt;% dplyr::filter(module == &quot;yellow&quot;) %&gt;% 
  readr::write_csv(here::here(&quot;output&quot;,&quot;yellow_genes.csv&quot;))</code></pre>
<div id="summary-output-of-network-analysis-result" class="section level3">
<h3>Summary output of network analysis result</h3>
<pre class="r"><code>## load biomart dataset
# ensembl &lt;- useMart(&quot;ensembl&quot;)
# ensembl_zebrafish &lt;- useDataset(&quot;drerio_gene_ensembl&quot;, mart = ensembl)
# write_rds(ensembl_zebrafish, here::here(&quot;analysis&quot;, &quot;ensembl_zebrafish.rds&quot;), compress = &quot;gz&quot;)
# allLLIDs &lt;- getBM(c(&quot;ensembl_gene_id&quot;, &quot;entrezgene_id&quot;, &quot;go_linkage_type&quot;),
#                   filters = c(&quot;ensembl_gene_id&quot;, &quot;with_entrezgene&quot;),
#                   values = list(&quot;ensembl_gene_id&quot; = names(X), &quot;with_entrezgene&quot;=TRUE),
#                   mart = ensembl_zebrafish) 
# write_rds(allLLIDs, here::here(&quot;analysis&quot;, &quot;allLLIDs.rds&quot;), compress = &quot;gz&quot;)
# allLLIDs &lt;- read_rds(here::here(&quot;analysis&quot;,&quot;allLLIDs.rds&quot;))

## Retain the genes with GO terms and reorder the rows to match the order of the genes in X.

# with_go_evidence &lt;- allLLIDs %&gt;%
#   distinct(ensembl_gene_id, .keep_all=TRUE) %&gt;%
#   column_to_rownames(&quot;ensembl_gene_id&quot;) %&gt;%
#   extract(names(X), ) %&gt;% 
#   rownames_to_column(&quot;ensembl_gene_id&quot;) %&gt;%
#   set_rownames(names(X))
# write_rds(with_go_evidence, here::here(&quot;analysis&quot;, &quot;with_go_evidence.rds&quot;), compress = &quot;gz&quot;)

with_go_evidence &lt;- read_rds(here::here(&quot;analysis&quot;,&quot;with_go_evidence.rds&quot;))</code></pre>
<pre class="r"><code>GOcollection &lt;- buildGOcollection(organism = &quot;zebrafish&quot;)</code></pre>
<pre><code> GOcollection: loading annotation data...
  ..preparing term lists... </code></pre>
<pre class="r"><code>GOenrichment &lt;- enrichmentAnalysis(
  classLabels = colors_zebrafish,
  identifiers = with_go_evidence$entrezgene_id,
  refCollection = GOcollection,
  useBackground = &quot;given&quot;,
  threshold = 1e-4,
  thresholdType = &quot;Bonferroni&quot;,
  getOverlapEntrez = TRUE,
  getOverlapSymbols = TRUE)</code></pre>
<pre><code> enrichmentAnalysis: preparing data..
  ..working on label set 1 ..</code></pre>
<pre class="r"><code>collectGarbage()

table.display &lt;- GOenrichment$enrichmentTable
table.display$overlapGenes &lt;- shortenStrings(table.display$overlapGenes, maxLength = 70, split = &quot;|&quot;)

write.csv(GOenrichment$enrichmentTable, here::here(&quot;output&quot;, &quot;GOenrichment_enrichmentTable.csv&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>devtools::session_info()</code></pre>
<pre><code>─ Session info ──────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.0 (2019-04-26)
 os       macOS Mojave 10.14.6        
 system   x86_64, darwin15.6.0        
 ui       X11                         
 language (EN)                        
 collate  en_AU.UTF-8                 
 ctype    en_AU.UTF-8                 
 tz       Australia/Adelaide          
 date     2020-10-28                  

─ Packages ──────────────────────────────────────────────────────────────
 package                            * version   date       lib
 acepack                              1.4.1     2016-10-29 [1]
 annotate                             1.62.0    2019-05-02 [1]
 AnnotationDbi                      * 1.46.1    2019-08-20 [1]
 anRichment                         * 1.18      2020-10-19 [1]
 anRichmentMethods                  * 0.92      2020-10-19 [1]
 assertthat                           0.2.1     2019-03-21 [1]
 backports                            1.1.4     2019-04-10 [1]
 base64enc                            0.1-3     2015-07-28 [1]
 Biobase                            * 2.44.0    2019-05-02 [1]
 BiocGenerics                       * 0.30.0    2019-05-02 [1]
 BiocParallel                         1.18.1    2019-08-06 [1]
 biomaRt                            * 2.40.4    2019-08-19 [1]
 Biostrings                           2.52.0    2019-05-02 [1]
 bit                                  1.1-14    2018-05-29 [1]
 bit64                                0.9-7     2017-05-08 [1]
 bitops                               1.0-6     2013-08-17 [1]
 blob                                 1.2.0     2019-07-09 [1]
 broom                                0.5.2     2019-04-07 [1]
 callr                                3.3.1     2019-07-18 [1]
 cellranger                           1.1.0     2016-07-27 [1]
 checkmate                            1.9.4     2019-07-04 [1]
 cli                                  1.1.0     2019-03-19 [1]
 cluster                              2.1.0     2019-06-19 [1]
 codetools                            0.2-16    2018-12-24 [1]
 colorspace                           1.4-1     2019-03-18 [1]
 crayon                               1.3.4     2017-09-16 [1]
 data.table                           1.12.2    2019-04-07 [1]
 DBI                                  1.0.0     2018-05-02 [1]
 DelayedArray                         0.10.0    2019-05-02 [1]
 desc                                 1.2.0     2018-05-01 [1]
 devtools                             2.2.2     2020-02-17 [1]
 digest                               0.6.20    2019-07-04 [1]
 doParallel                           1.0.15    2019-08-02 [1]
 dplyr                              * 0.8.3     2019-07-04 [1]
 dynamicTreeCut                     * 1.63-1    2016-03-11 [1]
 edgeR                              * 3.26.7    2019-08-13 [1]
 ellipsis                             0.3.0     2019-09-20 [1]
 evaluate                             0.14      2019-05-28 [1]
 fastcluster                        * 1.1.25    2018-06-07 [1]
 forcats                            * 0.4.0     2019-02-17 [1]
 foreach                              1.4.7     2019-07-27 [1]
 foreign                              0.8-72    2019-08-02 [1]
 Formula                              1.2-3     2018-05-03 [1]
 fs                                   1.3.1     2019-05-06 [1]
 genefilter                         * 1.66.0    2019-05-02 [1]
 generics                             0.0.2     2018-11-29 [1]
 GenomeInfoDb                       * 1.20.0    2019-05-02 [1]
 GenomeInfoDbData                     1.2.1     2019-07-25 [1]
 GenomicAlignments                    1.20.1    2019-06-18 [1]
 GenomicFeatures                    * 1.36.4    2019-07-09 [1]
 GenomicRanges                      * 1.36.0    2019-05-02 [1]
 ggplot2                            * 3.2.1     2019-08-10 [1]
 git2r                                0.26.1    2019-06-29 [1]
 glue                                 1.3.1     2019-03-12 [1]
 GO.db                              * 3.8.2     2019-08-09 [1]
 gridExtra                            2.3       2017-09-09 [1]
 gtable                               0.3.0     2019-03-25 [1]
 haven                                2.1.1     2019-07-04 [1]
 here                                 0.1       2017-05-28 [1]
 highr                                0.8       2019-03-20 [1]
 Hmisc                                4.2-0     2019-01-26 [1]
 hms                                  0.5.1     2019-08-23 [1]
 htmlTable                            1.13.1    2019-01-07 [1]
 htmltools                            0.3.6     2017-04-28 [1]
 htmlwidgets                          1.3       2018-09-30 [1]
 httpuv                               1.5.1     2019-04-05 [1]
 httr                                 1.4.1     2019-08-05 [1]
 impute                               1.58.0    2019-05-02 [1]
 IRanges                            * 2.18.2    2019-08-24 [1]
 iterators                            1.0.12    2019-07-26 [1]
 jsonlite                             1.6       2018-12-07 [1]
 kableExtra                         * 1.1.0     2019-03-16 [1]
 knitr                                1.24      2019-08-08 [1]
 labeling                             0.3       2014-08-23 [1]
 later                                0.8.0     2019-02-11 [1]
 lattice                              0.20-38   2018-11-04 [1]
 latticeExtra                         0.6-28    2016-02-09 [1]
 lazyeval                             0.2.2     2019-03-15 [1]
 limma                              * 3.40.6    2019-07-26 [1]
 locfit                               1.5-9.1   2013-04-20 [1]
 lubridate                            1.7.4     2018-04-11 [1]
 magrittr                           * 1.5       2014-11-22 [1]
 Matrix                               1.2-17    2019-03-22 [1]
 matrixStats                          0.54.0    2018-07-23 [1]
 memoise                              1.1.0     2017-04-21 [1]
 modelr                               0.1.5     2019-08-08 [1]
 munsell                              0.5.0     2018-06-12 [1]
 nlme                                 3.1-141   2019-08-01 [1]
 nnet                                 7.3-12    2016-02-02 [1]
 org.Dr.eg.db                       * 3.8.2     2019-11-20 [1]
 pander                             * 0.6.3     2018-11-06 [1]
 pheatmap                           * 1.0.12    2019-01-04 [1]
 pillar                               1.4.2     2019-06-29 [1]
 pkgbuild                             1.0.6     2019-10-09 [1]
 pkgconfig                            2.0.2     2018-08-16 [1]
 pkgload                              1.0.2     2018-10-29 [1]
 plyr                                 1.8.4     2016-06-08 [1]
 preprocessCore                       1.46.0    2019-05-02 [1]
 prettyunits                          1.0.2     2015-07-13 [1]
 processx                             3.4.1     2019-07-18 [1]
 progress                             1.2.2     2019-05-16 [1]
 promises                             1.0.1     2018-04-13 [1]
 ps                                   1.3.0     2018-12-21 [1]
 purrr                              * 0.3.3     2019-10-18 [1]
 R6                                   2.4.0     2019-02-14 [1]
 RColorBrewer                       * 1.1-2     2014-12-07 [1]
 Rcpp                                 1.0.2     2019-07-25 [1]
 RCurl                                1.95-4.12 2019-03-04 [1]
 readr                              * 1.3.1     2018-12-21 [1]
 readxl                               1.3.1     2019-03-13 [1]
 remotes                              2.1.1     2020-02-15 [1]
 reshape2                           * 1.4.3     2017-12-11 [1]
 rlang                                0.4.4     2020-01-28 [1]
 rmarkdown                            1.15      2019-08-21 [1]
 rpart                                4.1-15    2019-04-12 [1]
 rprojroot                            1.3-2     2018-01-03 [1]
 Rsamtools                            2.0.0     2019-05-02 [1]
 RSQLite                              2.1.2     2019-07-24 [1]
 rstudioapi                           0.10      2019-03-19 [1]
 rtracklayer                          1.44.3    2019-08-24 [1]
 rvest                                0.3.4     2019-05-15 [1]
 S4Vectors                          * 0.22.0    2019-05-02 [1]
 scales                               1.0.0     2018-08-09 [1]
 sessioninfo                          1.1.1     2018-11-05 [1]
 stringi                              1.4.3     2019-03-12 [1]
 stringr                            * 1.4.0     2019-02-10 [1]
 SummarizedExperiment                 1.14.1    2019-07-31 [1]
 survival                             2.44-1.1  2019-04-01 [1]
 testthat                             2.3.1     2019-12-01 [1]
 tibble                             * 2.1.3     2019-06-06 [1]
 tidyr                              * 0.8.3     2019-03-01 [1]
 tidyselect                           0.2.5     2018-10-11 [1]
 tidyverse                          * 1.2.1     2017-11-14 [1]
 TxDb.Hsapiens.UCSC.hg19.knownGene  * 3.2.2     2020-10-19 [1]
 TxDb.Mmusculus.UCSC.mm10.knownGene * 3.4.7     2020-10-19 [1]
 usethis                              1.5.1     2019-07-04 [1]
 vctrs                                0.2.0     2019-07-05 [1]
 viridisLite                          0.3.0     2018-02-01 [1]
 webshot                              0.5.1     2018-09-28 [1]
 WGCNA                              * 1.69      2020-02-28 [1]
 whisker                              0.4       2019-08-28 [1]
 withr                                2.1.2     2018-03-15 [1]
 workflowr                            1.6.0     2019-12-19 [1]
 xfun                                 0.9       2019-08-21 [1]
 XML                                  3.98-1.20 2019-06-06 [1]
 xml2                                 1.2.2     2019-08-09 [1]
 xtable                               1.8-4     2019-04-21 [1]
 XVector                              0.24.0    2019-05-02 [1]
 yaml                                 2.2.0     2018-07-25 [1]
 zeallot                              0.1.0     2018-01-28 [1]
 zlibbioc                             1.30.0    2019-05-02 [1]
 source        
 CRAN (R 3.6.0)
 Bioconductor  
 Bioconductor  
 local         
 local         
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 CRAN (R 3.6.0)
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 CRAN (R 3.6.0)
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  
 CRAN (R 3.6.0)
 CRAN (R 3.6.0)
 Bioconductor  

[1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library</code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
