---
title: "1_DE gene analysis"
author: "Yang & Steve"
date: "24/02/2020"
output: 
  html_document: 
    toc: yes
    toc_float: yes
    code_folding: hide
editor_options: 
  chunk_output_type: console
---
### Setup

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
```


```{r packages}
library(limma)
library(biomaRt)
library(edgeR)
library(AnnotationHub)
library(tidyverse)
library(magrittr)
library(pander)
library(RColorBrewer)
library(ggrepel)
library(swfdr)
library(scales)
library(here)
library(variancePartition)
library(org.Hs.eg.db)
library(plyr)
library(ggraph)
library(tidygraph)
library(RUVSeq)
library(GenomicRanges)
library(kableExtra)
library(org.Dr.eg.db)
theme_set(theme_bw())
panderOptions("big.mark", ",")
panderOptions("table.split.table", Inf)
panderOptions("table.style", "rmarkdown")
if (interactive()) setwd(here::here("analysis"))
```

```{R GC content & lengths information}
# Get GC content and lengths
gcTrans <- url("https://uofabioinformaticshub.github.io/Ensembl_GC/Release96/Danio_rerio.GRCz11.96.rds") %>%
  readRDS()
# Convert to the gene-level with average length, average GC, max length, and GC of longest transcript for all transcripts assigned to a single gene.
# gcGene <- gcTrans %>%
#   split(f = .$gene_id) %>%
#   mclapply(function(x){
#     gr <- reduce(x)
#     df <- DataFrame(
#       gene_id = unique(x$gene_id),
#       gene_symbol = unique(x$gene_symbol),
#       aveLength = round(mean(x$length),0) %>% as.integer(),
#       aveGc = sum(x$length * x$gc) / sum(x$length),
#       maxLength = max(x$length),
#       longestGc = x$gc[which.max(x$length)[[1]]]
#     )
#     mcols(gr) <- df
#     gr
#   }, mc.cores = 4) %>%
#   GRangesList() %>%
#   unlist() %>%
#   na.omit()
# x# This takes a reasonable amount of time
# # Save as .rds file for faster loading
# saveRDS(gcGene, file = "../R/gcGene.rds")
gcGene <- readRDS("../analysis/gcGene.rds")
```

Annotation was set up as a EnsDb object based on Ensembl Release 96

```{r ensDb, cache=TRUE}
ah <- AnnotationHub() %>%
  subset(species == "Danio rerio") %>%
  subset(dataprovider == "Ensembl") %>%
  subset(rdataclass == "EnsDb")
ensDb <- ah[["AH69169"]]
genes <- genes(ensDb)
transGR <- transcripts(ensDb)
mcols(genes) <- mcols(genes)[c("gene_id", "gene_name", 
                                   "gene_biotype", "entrezid","description","entrezid")] %>%
  as.data.frame() %>%
  dplyr::select(-entrezid) %>%
  left_join(as.data.frame(mcols(gcGene))) %>%
  distinct(gene_id, .keep_all = TRUE) %>%
  set_rownames(.$gene_id) %>%
  DataFrame() %>%
  .[names(genes),]
genesGR <- genes(ensDb)
```

Prior to count-level analysis, the initial dataset was pre-processed using the following steps:

- Adapters were removed from any reads
- Bases were removed from the end of reads when the quality score dipped below 30
- Reads < 35bp after trimming were discarded

After trimming alignment was performed using STAR v2.5.3a to the *Danio rerio* genome included in Ensembl Release 96 (GRCz11).
Aligned reads were counted for each gene if the following criteria were satisfied:

- Alignments were unique
- Alignments strictly overlapped exonic regions

```{r setThresholds}
minSamples <- 6
minCpm <- 1
```

```{r counts}
counts <- file.path("../data/2_alignedData/featureCounts/genes.out") %>%
	read_tsv() %>%
	set_colnames(basename(colnames(.))) %>%
	set_colnames(str_remove(colnames(.), "Aligned.+")) %>%
  gather(key = "Library", value = "Counts", -Geneid) %>%
  dplyr::mutate(Sample = str_remove_all(Library, "_2")) %>%
  group_by(Geneid, Sample) %>%
  dplyr::summarise(Counts = sum(Counts)) %>%
  tidyr::spread(key = "Sample", value = "Counts") %>%
  as.data.frame() %>%
  column_to_rownames("Geneid") 
genes2keep <- cpm(counts) %>%
  is_greater_than(minCpm) %>%
  rowSums() %>%
  is_weakly_greater_than(minSamples)
```

Counts were then loaded and summed across replicate sequencing runs.
Genes were only retained if receiving more than `r minCpm` cpm in at least `r minSamples` samples.
This filtering step discarded `r comma(sum(!genes2keep))` genes as undetectable and retained `r comma(sum(genes2keep))` genes for further analysis.

```{r plotDensities, fig.cap="Distribution of logCPM values for a) all and b) retained genes", fig.width = 4, fig.show='hold'}
plotDensities(cpm(counts, log = TRUE), legend = FALSE, main = "a) All genes")
plotDensities(cpm(counts[genes2keep,], log = TRUE), legend = FALSE, main = "b) Retained genes")
```

```{r dgeList}
dgeList <- counts %>%
  .[genes2keep,] %>%
  DGEList(
    samples = tibble(
      sample = colnames(.),
      group = str_replace_all(sample, "[0-9]*[A-Z]([12])Lardelli", "\\1"),
      pair = str_replace_all(sample, "[0-9]*([A-Z])[0-9].+", "\\1")
    ) %>%
      as.data.frame(),
		genes = genes[rownames(.)] %>%
			as.data.frame() %>%
			dplyr::select(
				ensembl_gene_id = gene_id,
				chromosome_name = seqnames,
				description,
				external_gene_name = gene_name,
				entrez_gene = entrezid.1,
				aveLength,
				aveGc,
				maxLength,
				longestGc
			)
	) %>%
	calcNormFactors(method = "TMM") %>%
	estimateDisp()
dgeList$samples$Genotype <- c("Q96K97del/+", "WT")[dgeList$samples$group] %>%
  factor(levels = c("WT", "Q96K97del/+"))
```

Counts were then formed into a DGEList object.
Library sizes after alignment, counting and gene filtering ranged between `r pander(comma(range(dgeList$samples$lib.size)))` reads.

# Data Inspection

```{r v}
v <- dgeList %>%
	voomWithQualityWeights(design = matrix(1, nrow = ncol(.)))
```

```{r plotWeights, echo=FALSE, fig.cap = "Sample weights with the ideal equal weight of 1 shown as a horizontal line."}
v$targets %>%
	ggplot(aes(sample, sample.weights, fill = Genotype)) +
	geom_bar(stat = "identity") +
	geom_hline(yintercept = 1, colour = "blue", linetype = 2) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))
```

```{r pca}
pca <- v$E %>%
    t() %>%
    prcomp() 
summary(pca)$importance %>% pander(split.tables = Inf)    
```

```{r plotPCA, echo=FALSE, fig.cap = "PCA of all samples. Point sizes indicate library sizes with the combination of PC1 & PC2 appearing to capture a considerable amount of this variation."}
pca$x %>%
	as.data.frame() %>%
	rownames_to_column("sample") %>%
	left_join(v$targets) %>%
	ggplot(aes(PC1, PC2, colour = Genotype)) +
	geom_point(aes(size = sample.weights)) +
	geom_text_repel(aes(label = sample), show.legend = FALSE) +
	labs(size = "Sample weight") +
  xlab(paste0("PC1 (", percent(summary(pca)$importance[2,"PC1"]), ")")) +
  ylab(paste0("PC2 (", percent(summary(pca)$importance[2,"PC2"]), ")")) +
	theme(legend.title.align = 0.5)
```

#DGE Analysis

## Setup a design matrix using each pool as its own intercept, with the genotype as the common difference
```{r}
expDesign <- model.matrix(~0 + pair + Genotype, dgeList$samples) %>%
  set_colnames(str_replace(colnames(.), pattern = "Genotype.+", "Mutant"))
```

## Recalculate the dispersions
```{r}
dgeList %<>%
  estimateGLMCommonDisp(expDesign) %>%
  estimateGLMTagwiseDisp(expDesign)
```

```{r deGenes}
deGenes <- dgeList %>%
  glmFit(expDesign) %>%
  glmLRT(coef = "Mutant") %>%
  topTags(n = Inf) %>%
  as.data.frame() %>%
  set_colnames(gsub("ID.", "", colnames(.))) %>%
  as_tibble() %>%
  mutate(
    Sig = FDR < 0.05,
    rankstat = -sign(logFC)*log10(PValue)
  ) %>%
  dplyr::select(
    ensembl_gene_id, external_gene_name, logFC, logCPM, LR,
  PValue, FDR, Sig,aveGc, aveLength, rankstat
  ) %>%
  arrange(PValue)
deGenesSig <- deGenes %>%
  dplyr::filter(Sig)
```


```{r volcano, echo=FALSE, fig.cap = "Volcano plot highlighting DE genes. Genes indicated in red belowd were considered as DE, which receive a FDR < 0.05."}
deGenes %>%
  ggplot(aes(logFC, -log10(PValue), colour = Sig)) +
  geom_point() +
  geom_text_repel(aes(label = external_gene_name), 
					data = . %>% dplyr::filter(Sig & abs(logFC) > 1.2)) +
  geom_vline(xintercept = c(-1, 1)*0.5, linetype = 2, colour = "grey50") +
  geom_hline(yintercept = deGenes %>%
			   	dplyr::filter(FDR < 0.05) %>%
			   	.[["P.Value"]] %>%
			   	max() %>% 
			   	log(10) %>%
			   	multiply_by(-1), 
			   linetype = 2, colour = "blue") +
  scale_colour_manual(values = c(rgb(0.5, 0.5, 0.5, 0.5), "red")) +
	scale_x_continuous(breaks = seq(-6, 6, by = 2)) +
	theme(legend.position = "none")
```

```{r plotMA, echo=FALSE, fig.cap="logFC plotted against expression level with significant DE genes shown in red."}
deGenes %>%
	ggplot(aes(logCPM, logFC)) +
	geom_point(aes(colour = Sig), alpha = 0.5) +
	geom_text_repel(aes(label = external_gene_name, colour = Sig), 
					data = . %>% dplyr::filter(Sig & abs(logFC) > 1.2)) +
  geom_smooth(se = FALSE) +
	geom_hline(yintercept = c(-1, 1)*0.5, linetype = 2, colour = "grey50") +
	labs(
		x = "Average Expression (logCPM)",
		y = "logFC"
	) +
	scale_colour_manual(values = c(rgb(0.5, 0.5, 0.5, 0.5), "red")) +
	theme(legend.position = "none")
```

# RUV treatment (Remove Unwanted Variation)
# Find some 'unchanged' genes. Grab the lowest ranked 5000
```{r}
genes4Control <- deGenes %>%
  arrange(desc(PValue)) %>%
  dplyr::slice(1:5000) %>%
  .[["ensembl_gene_id"]]
length(genes4Control)
```

```{r RUV}
k <- 1
ruv <- RUVg(
  dgeList$counts, 
  cIdx = genes4Control,
  k = k, 
  isLog = FALSE,
  round = TRUE
)
dgeList$samples <- cbind(dgeList$samples, ruv$W)
```

```{r plotPCA 2}
pcaRUV <- ruv$normalizedCounts %>%
  cpm(log = TRUE) %>%
  t() %>%
  prcomp() 
summary(pcaRUV)$importance %>% pander(split.tables = Inf)    
pcaRUV$x %>%
  cbind(dgeList$samples[rownames(.),]) %>%
  ggplot(aes(PC1, PC2, colour = Genotype)) +
  geom_point() +
  stat_ellipse(aes(fill = Genotype), geom = "polygon", alpha = 0.2) +
  xlab(
    paste0(
      "PC1 (", percent(summary(pcaRUV)$importance[2, "PC1"]), ")"
    )
  ) +
  ylab(
    paste0(
      "PC2 (", percent(summary(pcaRUV)$importance[2, "PC2"]), ")"
    )
  )
```

```{r}
expDesignRUV <- model.matrix(~0 + pair + Genotype + W_1, dgeList$samples) %>%
  set_colnames(str_replace(colnames(.), pattern = "Genotype.+", "Mutant"))
```

## Recalculate the dispersions
```{r}
dgeList %<>%
  estimateGLMCommonDisp(expDesignRUV) %>%
  estimateGLMTagwiseDisp(expDesignRUV)
```

## Now run DE Analysis after RUV
```{r topTable}
topTable <- dgeList %>%
  glmFit(expDesignRUV) %>%
  glmLRT(coef = "Mutant") %>%
  topTags(n = Inf) %>%
  as.data.frame() %>%
  set_colnames(gsub("ID.", "", colnames(.))) %>%
  as_tibble() %>%
  mutate(
    DE = FDR < 0.01,
    rankstat = -sign(logFC)*log10(PValue)
  ) %>%
  dplyr::select(
    ensembl_gene_id, external_gene_name, logFC, logCPM, LR,
  PValue, FDR, DE,aveGc, aveLength, rankstat
  ) %>%
  arrange(PValue)
topTableDE <- topTable %>%
  dplyr::filter(DE == TRUE)
```

## Check the MA plot
```{r plotMA 2}
tested <- c("si:ch211-213a13.2", "si:ch211-11p18.6", "mdh1ab", "CABZ01034698.2")
# topTable %>%
topTable %>%
  ggplot(aes(logCPM, logFC)) +
  geom_point(aes(colour = DE), alpha = 0.5) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = c(-1, 1)*0.5, linetype = 2) +
  geom_text_repel(
    aes(label = external_gene_name, colour = DE),
    data = . %>%
      dplyr::filter(abs(logFC) > 1.2 & DE | external_gene_name %in% tested),
    show.legend = FALSE
  ) +
  scale_colour_manual(values = c("grey50", "red"))
```

## Check the volcano plot
```{r plotvolcano}
topTable %>%
  mutate(DE = FDR < 0.01) %>%
  ggplot(aes(logFC, -log10(PValue))) +
  geom_point(aes(colour = DE), alpha = 0.5) +
  geom_vline(xintercept = c(-1, 1)*0.5, linetype = 2) +
  geom_text_repel(
    aes(label = external_gene_name, colour = DE),
    data = . %>%
      dplyr::filter(abs(logFC) > 1.2 & DE | external_gene_name %in% tested),
    show.legend = FALSE
  ) +
  scale_colour_manual(values = c("grey50", "red"))
```

## Compare the two lists
```{r}
topTable %>%
  dplyr::select(ensembl_gene_id, RUV = logFC, DE) %>%
  left_join(deGenes) %>%
  dplyr::rename(Raw = logFC) %>%
  mutate(
    Status = case_when(
      Sig & DE ~ "Both",
      Sig & !DE ~ "Raw Only",
      !Sig & DE ~ "RUV Only",
      !Sig & !DE ~ "Not DE"
    )
  ) %>%
  ggplot(aes(Raw, RUV,)) +
  geom_point(aes(colour = Status), alpha = 0.5) +
  geom_abline(slope= 1, linetype = 2) +
  geom_vline(xintercept = c(-1, 1)*0.5) +
  geom_hline(yintercept = c(-1, 1)*0.5) +
  scale_colour_manual(values = c("green", "grey50", "red", "blue"))
```

The `r sum(topTable$DE)` genes considered as DE using an FDR of 0.01 and logFC beyond the range $\pm 0.5$ were then inspected to confirm that the counts support their inclusion as DE.

Here plot CPM for DE genes having a FDR below 0.01 and logFC beyond the range $\pm 1.2$

```{r plotCPM, echo=FALSE, fig.cap="Expression values for each of the potential DE genes using raw CPM values and a log-scale y-axis."}
dgeList %>%
	cpm(log = TRUE) %>%
	.[dplyr::filter(topTable, FDR < 0.01 & abs(logFC) > 1.2)$ensembl_gene_id,] %>%
	as.data.frame() %>%
	rownames_to_column("ensembl_gene_id") %>%
	as_tibble() %>%
	gather(key = "sample", value = "CPM", -ensembl_gene_id) %>%
	left_join(dgeList$samples) %>%
	left_join(dgeList$genes) %>%
	ggplot(aes(group, CPM)) +
	geom_boxplot(aes(fill = group)) +
	geom_jitter(width = 0.1) +
	facet_wrap(~external_gene_name, scales = "free_y") +
	theme(legend.position = "none")
```


# Data Export

Final gene lists were exported as separate csv files, along with dgeList objects.

```{r writeResults}
write.csv(topTableDE,"../output/DEgenes.csv")
write.csv(topTable,"../output/topTable.csv")
write_rds(dgeList, "../data/dgeList.rds", compress = "gz")
```

# Session Info

```{r}
pander(sessionInfo())
```