---
title: "5_WGCNA"
author: "Yang & Steve"
date: "29/09/2020"
output: 
  html_document: 
    toc: yes
    toc_float: yes
    code_folding: hide
editor_options: 
  chunk_output_type: console
---
### Setup

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
```

```{r packages}
library(dplyr)
library(magrittr)
library(reshape2)
library(readr)
library(tibble)
library(tidyverse)
library(pander)
library(biomaRt)
library(limma)
library(edgeR)
library(genefilter)
library(WGCNA)
library(dynamicTreeCut)
library(ggplot2)
library(RColorBrewer)
library(grid)
  # Viewpoints for showing 4 plots on a 2x2 grid
  vp1 <- viewport(x = 0, y = 0, width = 0.5, height = 0.5, just = c(0, 0))
  vp2 <- viewport(x = 0.5, y = 0, width = 0.5, height = 0.5, just = c(0,0))
  vp3 <- viewport(x = 0, y = 0.5, width = 0.5, height = 0.5, just = c(0, 0))
  vp4 <- viewport(x = 0.5, y = 0.5, width = 0.5, height = 0.5, just = c(0, 0))
library(kableExtra)
library(pheatmap)
library(anRichment)

theme_set(theme_bw())
panderOptions("big.mark", ",")
panderOptions("table.split.table", Inf)
panderOptions("table.style", "rmarkdown")
if (interactive()) setwd(here::here("analysis"))
```


# Data load

```{r loaddgeList}
dgeList <- read_rds(here::here("data","dgeList.rds"))
entrezGenes <- dgeList$genes %>%
  dplyr::filter(!is.na(entrez_gene)) %>%
  unnest(entrez_gene) %>%
  dplyr::rename(entrez_gene = entrez_gene)
topTable <- file.path(here::here("output", "topTable.csv")) %>% 
  read_csv()
```

# Filtering

Only keep the genes of high expression (cpm > 1.5 in 6 samples) and high variation (not in the bottom 25%).

```{r}
A <- dgeList %>% 
  cpm(log = TRUE) %>% 
  melt %>% 
  dplyr::filter(is.finite(value)) %>% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle("Before filtering") +
  labs(x = "logCPM", y = "Density")

# Retain genes which are expressed at sufficiently high levels, at least 1.5 cpm in 6 (half of the total number of) samples.
keepTheseGenes <- rowSums(cpm(dgeList) > 1.5) >= 6  

# Density plot of the gene expression distribution after filtering out low expressed genes. 
B <- dgeList %>% 
  cpm(log = TRUE) %>% 
  .[keepTheseGenes,] %>%
  melt %>% 
  dplyr::filter(is.finite(value)) %>% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle("After filtering low exp genes")

# Retain genes with sufficiently high variation in expression across all samples.
# Here we define sufficiently high variation as having a standard deviation above the 25th quantile (so not in bottom 25%).
variableGenes <- rowSds(cpm(dgeList, log=TRUE)) >  quantile(rowSds(cpm(dgeList, log=TRUE)), 0.75)

# Density plot of gene expression distribution after filtering out low variance genes. 
C <- dgeList %>% 
  cpm(log = TRUE) %>% 
  .[variableGenes,] %>%
  melt %>% 
  dplyr::filter(is.finite(value)) %>% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle("After filtering variable genes")

# Density plot of gene expression distribution after filtering out both low expressed and low variance genes. 
D <- dgeList %>% 
  cpm(log = TRUE) %>% 
  .[keepTheseGenes&variableGenes,] %>%
  melt %>% 
  dplyr::filter(is.finite(value)) %>% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle("After filtering both")

# Plot the effect of filtering low expressed genes, low variation genes, and both.
grid.newpage()
print(A, vp = vp3)
print(B, vp = vp4)
print(C, vp = vp1)
print(D, vp  = vp2)

dgeList_filtered <- dgeList[keepTheseGenes&variableGenes,,keep.lib.sizes = FALSE]
```

There are `r dim(dgeList$counts)[1]` in dgeList. After filtering, `r dim(dgeList_filtered$counts)[1]` genes are retained. 

# Convert gene counts into logCPM form and stored in gene count matraix `X`. 
```{r Transform and Transpose Matrix, message=FALSE, warning=FALSE}
X <- dgeList_filtered$counts %>% 
  cpm(log=TRUE) %>% t %>% as.data.frame
```

## Check Sample Clustering

Here we use the gene expression patterns described in `X` to build sample dendrogram (sample tree), and check how the `traits` related to the sample dendrogram.

`traits` uses the design matrx used in previous DE genes analysis. 

```{r Check Sample Clustering, warning=FALSE, message=FALSE, fig.cap = "Hierarchial clustering of Q96K97del larvae RNA-seq libraries based on expression of 13,538 genes in the matrix X. "}
# Apply hierarchial clustering to X. 
sampleTree <- hclust(dist(X), method = "average")

# Using the design matrix, which contains information of samples' pair and genotype
traits <- expDesign <- model.matrix(~0 + pair + Genotype, dgeList$samples) %>%
  set_colnames(str_replace(colnames(.), pattern = "Genotype.+", "Mutant")) %>% as.data.frame

# Assign each group in design matrix to a colour. 
traitColors <- numbers2colors(traits, signed = FALSE, colors = c("#ffffff", "#73626E"))

# Plot dendrogram of clustered samples with the design matrix underneath. 
plotDendroAndColors(sampleTree, traitColors,
                    groupLabels = names(traits),
                    main = "Sample dendrogram and trait heatmap")
```


### Choosing the soft-Thresholding Power

Constructing a weighted gene network requires a soft-thresholding power Î², which will be chosen based on the scale-free topology criterion described by [Zhang & Horvath (2005)](https://doi.org/10.2202/1544-6115.1128). 

Here we performs the analysis of network topology for choosing a proper soft-thresholding power.

The following plots show the sufficiently high scale-free topology index (R<sub>2</sub>). We choose 13 as soft thresholding power, which is the lowest power reaching a high value (roughly 0.8).

```{r Soft Thresholding Power Plots, warning=FALSE, message=FALSE, cache=TRUE, fig.cap="Analysis of network topology for various soft-thresholding powers."}
# Choose a set of soft thresholding powers and call the network topology analysis fucntion.
# Here I use a signed network 
powers <- c(c(1:10), seq(from = 12, to = 26, by = 2))
sft <- pickSoftThreshold(X, powerVector = powers, verbose = 5, networkType = "signed")

# Plot the effect of soft-thresholding power on network topologies. 
sizeGrWindow(9, 5)
par(mfrow = c(1,2))
cex1 = 0.9
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2], labels=powers,cex=cex1,col="red")
# this line corresponds to using an R^2 cut-off of 0.80.
abline(h=0.8,col="red")

plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
```

Choose the power 13.

## Step by step network construction

```{r adjacency, TOM, and Clustering using TOM}
softPower <- 13

# Calculate the adjacency matrix, using the soft thresholding power 13
adjacency_zebrafish <- adjacency(X, power = softPower, type = "signed")

# To minimise effects of noise and spurious correlations, we transform the adjacency into a Topological Overlap Matrix (TOM) and calculate the dissimilarity.
TOM_zebrafish <- TOMsimilarity(adjacency_zebrafish, TOMType = "signed")
dissTOM_zebrafish <- 1-TOM_zebrafish

# We now use hierarchial clustering to produce a hierarchial clustering tree (dendrogram) of genes using TOM
geneTree_zebrafish <- hclust(as.dist(dissTOM_zebrafish), method = "average")

# Plot the dendrogram
plot(geneTree_zebrafish, xlab = "", ylab = "", main = "Gene clustering on TOM-based dissimilarity", labels = FALSE, hang = 0.04)
```


## Check similarity of modules before merging

```{r}
dynamicMods = cutreeDynamic(dendro = geneTree_zebrafish, distM = dissTOM_zebrafish,
              deepSplit = 2, pamRespectsDendro = FALSE,
              minClusterSize = 50)
dynamicColors = labels2colors(dynamicMods)
table(dynamicColors)
MEList = moduleEigengenes(X, colors = dynamicColors)
MEs = MEList$eigengenes
# Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs);
# Cluster module eigengenes
METree = hclust(as.dist(MEDiss), method = "average");
# Plot the result
sizeGrWindow(7, 6)
plot(METree, main = "Clustering of module eigengenes",
xlab = "", sub = "")
```



## Final modules

```{r Optimised Initial Dendrogram}
# Using 80 as minimum module size With PAM
treeCut_zebrafish <- dynamicTreeCut::cutreeDynamic(
  dendro = geneTree_zebrafish,
  distM = dissTOM_zebrafish,
  method = "hybrid",
  minClusterSize = 80,
  minGap = 0.25,
  maxCoreScatter = 0.75,
  verbose = 0,
  pamStage = TRUE,
  #deepSplit = 3,
  cutHeight = 0.8
)

colors_zebrafish <- labels2colors(treeCut_zebrafish)
table(colors_zebrafish)

plotDendroAndColors(geneTree_zebrafish, 
                    colors_zebrafish, "Dynamic Tree Cut",
                    dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05,
                    main = "Gene dendrogram and module colours")


```


### Relating modules to `traits`

```{r Correlation between Modules and Groups, message=FALSE, warning=FALSE, fig.cap="Heatmap showing Spearman correlation between identified network modules and biological groups.", fig.height=8}
# Define numbers of genes and samples
nGenes <- ncol(X)
nSamples <- nrow(X)

# Recalculate the MEs with colour labels
MEs0 <- moduleEigengenes(X, colors_zebrafish)$eigengenes
MEs <- orderMEs(MEs0)


moduleTraitCor <- cor(MEs, traits, use = "p", method = "spearman")
# Remove the grey module
moduleTraitCor <- moduleTraitCor[rownames(moduleTraitCor) != "MEgrey",]
#moduleTraitCor <- bicor(MEs, traits, use = "p", maxPOutliers = 0.1, robustY=FALSE)
moduleTraitPValue <- corPvalueStudent(moduleTraitCor, nSamples)
```

### Plot the relationships between module and `trait`

```{r}
# Visualise relationships between modules and traits.
# Will display correlations and their p-values
textMatrix <- paste(signif(moduleTraitCor, 2), "\n(", signif(moduleTraitPValue, 1), ")", sep = "")
dim(textMatrix) <- dim(moduleTraitCor)

pheatmap::pheatmap(
  moduleTraitCor,
  color = colorRampPalette(c("#10c1e5", "#82e0b4","#F9F9F9", "#FBB829", "#FF0066"))(100), 
  labels_row = gsub(x = names(MEs), pattern = "ME", replacement = ""),
  border_color = "white", 
  treeheight_row = 0, treeheight_col = 0,
  clustering_distance_rows = "correlation",
  clustering_distance_cols = "correlation", 
  cutree_cols = 3, 
  cluster_cols = FALSE, number_color = "black",
  cellwidth = 25, cellheight = 25,fontsize = 9,
  display_numbers = textMatrix 
  )
```


```{r Table-with-number-of-genes-per-module}
genesInModule <- data.frame(module = colors_zebrafish, gene = colnames(X))

colors_zebrafish %>% table %>% as.data.frame %>% set_colnames(c("Module", "Genes")) %>% dplyr::arrange(desc(Genes)) %>% kable(format="html")
```

## genes in each module

```{r genes in each Module, eval=FALSE}
genesInModule %>% dplyr::filter(module == "yellow") %>% 
  readr::write_csv(here::here("output","yellow_genes.csv"))

```

### Summary output of network analysis result

```{r get entrez gene IDs}
## load biomart dataset
# ensembl <- useMart("ensembl")
# ensembl_zebrafish <- useDataset("drerio_gene_ensembl", mart = ensembl)
# write_rds(ensembl_zebrafish, here::here("analysis", "ensembl_zebrafish.rds"), compress = "gz")
# allLLIDs <- getBM(c("ensembl_gene_id", "entrezgene_id", "go_linkage_type"),
#                   filters = c("ensembl_gene_id", "with_entrezgene"),
#                   values = list("ensembl_gene_id" = names(X), "with_entrezgene"=TRUE),
#                   mart = ensembl_zebrafish) 
# write_rds(allLLIDs, here::here("analysis", "allLLIDs.rds"), compress = "gz")
# allLLIDs <- read_rds(here::here("analysis","allLLIDs.rds"))

## Retain the genes with GO terms and reorder the rows to match the order of the genes in X.

# with_go_evidence <- allLLIDs %>%
#   distinct(ensembl_gene_id, .keep_all=TRUE) %>%
#   column_to_rownames("ensembl_gene_id") %>%
#   extract(names(X), ) %>% 
#   rownames_to_column("ensembl_gene_id") %>%
#   set_rownames(names(X))
# write_rds(with_go_evidence, here::here("analysis", "with_go_evidence.rds"), compress = "gz")

with_go_evidence <- read_rds(here::here("analysis","with_go_evidence.rds"))
```

```{r GO enrichment analysis}
GOcollection <- buildGOcollection(organism = "zebrafish")
GOenrichment <- enrichmentAnalysis(
  classLabels = colors_zebrafish,
  identifiers = with_go_evidence$entrezgene_id,
  refCollection = GOcollection,
  useBackground = "given",
  threshold = 1e-4,
  thresholdType = "Bonferroni",
  getOverlapEntrez = TRUE,
  getOverlapSymbols = TRUE)

collectGarbage()

table.display <- GOenrichment$enrichmentTable
table.display$overlapGenes <- shortenStrings(table.display$overlapGenes, maxLength = 70, split = "|")

write.csv(GOenrichment$enrichmentTable, here::here("output", "GOenrichment_enrichmentTable.csv"))

```





