---
title: "RNA-seq analysis"
author: "Yang"
date: "01/08/2019"
output: html_document
editor_options: 
  chunk_output_type: console
---
## Setup

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
```

```{r packages}
library(limma)
library(edgeR)
library(AnnotationHub)
library(tidyverse)
library(magrittr)
library(pander)
library(RColorBrewer)
library(ggrepel)
library(swfdr)
library(qvalue)
library(scales)
library(here)
theme_set(theme_bw())
panderOptions("big.mark", ",")
panderOptions("table.split.table", Inf)
panderOptions("table.style", "rmarkdown")
if (interactive()) setwd(file.path(here(), "R"))
```

Annotation was set up as a EnsDb object based on Ensembl Release 96

```{r ensDb, cache=TRUE}
ah <- AnnotationHub() %>%
  subset(species == "Danio rerio") %>%
  subset(dataprovider == "Ensembl") %>%
  subset(rdataclass == "EnsDb")
ensDb <- AnnotationHub()[["AH69169"]]
genes <- genes(ensDb)
```
Prior to count-level analysis, the initial dataset was pre-processed using the following steps:

- Adapters were removed from any reads
- Bases were removed from the end of reads when the quality score dipped below 30
- Reads < 35bp after trimming were discarded

After trimming alignment was performed using STAR v2.5.3a to the *Danio rerio* genome included in Ensembl Release 96 (GRCz11).
Aligned reads were counted for each gene if the following criteria were satisfied:

- Alignments were unique
- Alignments strictly overlapped exonic regions

```{r setThresholds}
minSamples <- 6
minCpm <- 1
```


```{r counts}
counts <- file.path("../2_alignedData/featureCounts/genes.out") %>%
	read_tsv() %>%
	set_colnames(basename(colnames(.))) %>%
	set_colnames(str_remove(colnames(.), "Aligned.+")) %>%
  gather(key = "Library", value = "Counts", -Geneid) %>%
  mutate(Sample = str_remove_all(Library, "_2")) %>%
  group_by(Geneid, Sample) %>%
  summarise(Counts = sum(Counts)) %>%
  spread(key = "Sample", value = "Counts") %>%
  as.data.frame() %>%
  column_to_rownames("Geneid") 
genes2keep <- cpm(counts) %>%
  is_greater_than(minCpm) %>%
  rowSums() %>%
  is_weakly_greater_than(minSamples)
```

Counts were then loaded and summed across replicate sequencing runs.
Genes were only retained if receiving more than `r minCpm` cpm in at least `r minSamples` samples.
This filtering step discarded `r comma(sum(!genes2keep))` genes as undetectable and retained `r comma(sum(genes2keep))` genes for further analysis.

```{r plotDensities, fig.cap="Distribution of logCPM values for a) all and b) retained genes", fig.width = 4, fig.show='hold'}
plotDensities(cpm(counts, log = TRUE), legend = FALSE, main = "a) All genes")
plotDensities(cpm(counts[genes2keep,], log = TRUE), legend = FALSE, main = "b) Retained genes")
```


```{r dgeList}
dgeList <- counts %>%
  .[genes2keep,] %>%
  DGEList(
    samples = tibble(
      sample = colnames(.),
      group = str_replace_all(sample, "[0-9]*[A-Z]([12])Lardelli", "\\1")
    ) %>%
      as.data.frame(),
		genes = genes[rownames(.)] %>%
			as.data.frame() %>%
			dplyr::select(
				ensembl_gene_id = gene_id,
				chromosome_name = seqnames,
				description,
				external_gene_name = gene_name
			)
	) %>%
	calcNormFactors(method = "TMM") %>%
	estimateDisp()
dgeList$samples$Genotype <- c("Q96K97del/+", "WT")[dgeList$samples$group] %>%
  factor(levels = c("WT", "Q96K97del/+"))
```

Counts were then formed into a DGEList object.
Library sizes after alignment, counting and gene filtering ranged between `r pander(comma(range(dgeList$samples$lib.size)))` reads.

## Data Inspection
```{r v}
v <- dgeList %>%
	voomWithQualityWeights(design = matrix(1, nrow = ncol(.)))
```

```{r plotWeights, echo=FALSE, fig.cap = "Sample weights with the ideal equal weight of 1 shown as a horizontal line."}
v$targets %>%
	ggplot(aes(sample, sample.weights, fill = Genotype)) +
	geom_bar(stat = "identity") +
	geom_hline(yintercept = 1, colour = "blue", linetype = 2) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))
```

```{r pca}
pca <- v$E %>%
    t() %>%
    prcomp() 
summary(pca)$importance %>% pander(split.tables = Inf)    
```

```{r plotPCA, echo=FALSE, fig.cap = "PCA of all samples. Point sizes indicate library sizes with the combination of PC1 & PC2 appearing to capture a considerable amount of this variation."}
pca$x %>%
	as.data.frame() %>%
	rownames_to_column("sample") %>%
	left_join(v$targets) %>%
	ggplot(aes(PC1, PC2, colour = Genotype)) +
	geom_point(aes(size = sample.weights)) +
	geom_text_repel(aes(label = sample), show.legend = FALSE) +
	labs(size = "Sample weight") +
	theme(legend.title.align = 0.5)
```

psen1 expression checking

```{r plotpsen1, echo=FALSE, fig.cap = "Expression of *psen1* across all samples."}
dgeList %>%
	cpm(log = TRUE) %>%
	.["ENSDARG00000004870",] %>%
	as.matrix() %>%
	set_colnames("logCPM") %>%
	cbind(dgeList$samples[rownames(.),]) %>%
	ggplot(aes(sample, logCPM, fill = Genotype)) +
	geom_bar(stat = "identity") +
  facet_wrap(~Genotype, scales = "free_x") +
  guides(fill = FALSE) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))
```

rpl13 expression checking (house-keeping gene)

```{r plotrpl13, echo=FALSE, fig.cap = "Expression of *rpl13* across all samples."}
dgeList %>%
	cpm(log = TRUE) %>%
	.["ENSDARG00000099380",] %>%
	as.matrix() %>%
	set_colnames("logCPM") %>%
	cbind(dgeList$samples[rownames(.),]) %>%
	ggplot(aes(sample, logCPM, fill = Genotype)) +
	geom_bar(stat = "identity") +
  facet_wrap(~Genotype, scales = "free_x") +
  guides(fill = FALSE) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))
```

## Analysis

### DGE Analysis

```{r efit}
design <- model.matrix(~Genotype, data = v$targets)
colnames(design) <- c("Wildtype", "MutvsWildtype")
efit <- v %>%
	lmFit(design) %>%
	eBayes()
```

```{r deGenes}
deGenes <- efit %>%
	topTable(coef = "MutvsWildtype", number = Inf, sort.by = "p") %>%
	as_tibble() %>%
  mutate(
    q = qvalue(P.Value)$qvalues,
    Sig = q < 0.1 & abs(logFC) > 0.9
  )
```


```{r volcano, echo=FALSE, fig.cap = "Volcano plot highlighting DE genes. Genes were considered as DE if receiving a q-value < 0.05 and with logFC beyond the range $\\pm 0.9$. Genes indicated in red below the blue line recieved a q-value < 0.1 and are of potential interest."}
deGenes %>%
	ggplot(aes(logFC, -log10(P.Value), colour = Sig)) +
	geom_point() +
	geom_text_repel(aes(label = external_gene_name), 
					data = . %>% dplyr::filter(Sig)) +
	geom_vline(xintercept = c(-1, 1)*0.9, linetype = 2, colour = "grey50") +
	geom_hline(yintercept = deGenes %>%
			   	dplyr::filter(q < 0.05) %>%
			   	.[["P.Value"]] %>%
			   	max() %>% 
			   	log(10) %>%
			   	multiply_by(-1), 
			   linetype = 2, colour = "blue") +
	scale_colour_manual(values = c(rgb(0.5, 0.5, 0.5, 0.5), "red")) +
	scale_x_continuous(breaks = seq(-6, 6, by = 2)) +
	theme(legend.position = "none")
```

```{r plotMA, echo=FALSE, fig.cap="logFC plotted against expression level with significant DE genes shown in red."}
deGenes %>%
	ggplot(aes(AveExpr, logFC, colour = Sig)) +
	geom_point() +
	geom_text_repel(aes(label = external_gene_name), 
					data = . %>% dplyr::filter(Sig | abs(logFC) > 3.5)) +
	geom_hline(yintercept = c(-1, 1)*0.9, linetype = 2, colour = "grey50") +
	labs(
		x = "Average Expression (logCPM)",
		y = "logFC"
	) +
	scale_colour_manual(values = c(rgb(0.5, 0.5, 0.5, 0.5), "red")) +
	theme(legend.position = "none")
```

The `r sum(deGenes$Sig)` genes considered as DE using an FDR of 0.05 and logFC beyond the range $\pm 0.9$ were then inspected to confirm that the counts support their inclusion as DE.


```{r plotCPM, echo=FALSE, fig.cap="Expression values for each of the potential DE genes using raw CPM values and a log-scale y-axis."}
dgeList %>%
	cpm(log = TRUE) %>%
	.[dplyr::filter(deGenes, Sig)$ensembl_gene_id,] %>%
	as.data.frame() %>%
	rownames_to_column("ensembl_gene_id") %>%
	as_tibble() %>%
	gather(key = "sample", value = "CPM", -ensembl_gene_id) %>%
	left_join(dgeList$samples) %>%
	left_join(dgeList$genes) %>%
  mutate(external_gene_name = case_when(
    external_gene_name %in% dplyr::filter(deGenes, q < 0.05)$external_gene_name ~ paste0(external_gene_name, "**"),
    !external_gene_name %in% dplyr::filter(deGenes, q < 0.05)$external_gene_name ~ external_gene_name
  )) %>%
	ggplot(aes(group, CPM)) +
	geom_boxplot(aes(fill = group)) +
	geom_jitter(width = 0.1) +
	facet_wrap(~external_gene_name, scales = "free_y") +
	# scale_y_log10() +
	theme(legend.position = "none")
```

```{r, echo=FALSE}
deGenes %>%
	dplyr::slice(1:20) %>%
	dplyr::select(-chromosome_name, -B, -Sig) %>%
  dplyr::rename(FDR = adj.P.Val) %>%
	mutate(description = gsub("([^\\[]*).+", "\\1", description)) %>%
	pander(
		justify = "lllrrrrrr",
		caption = "Top 20 results from DGE analysis. FDR indicates the Benjamini-Hochberg FDR, whilst q-values represent the Storey FDR.")
```


The final results were exported as the file `Q96K97delhetvsWildtype.csv`

```{r exportCSV}
write_csv(deGenes, "../results/Q96K97delhetvsWildtype.csv")
```

## Enrichment Analysis

An existing mapping of zebrafish Ensembl identifiers to human Entrez identifiers was obtained

```{r ens2Entrez}
ens2Entrez <- file.path("../R/mart_export.tsv") %>% 
    read_tsv()
goSummaries <- url("https://uofabioinformaticshub.github.io/summaries2GO/data/goSummaries.RDS") %>%
  readRDS()
```

```{r}
de <- deGenes %>%
    dplyr::filter(q < 0.1 & abs(logFC) > 0.9) %>%
    dplyr::select(`Gene stable ID` = ensembl_gene_id) %>%
    left_join(ens2Entrez) %>%
    dplyr::filter(!is.na(`NCBI gene ID`)) %>%
    .[["NCBI gene ID"]] %>%
    unique()
uv <- deGenes %>%
    dplyr::select(`Gene stable ID` = ensembl_gene_id) %>%
    left_join(ens2Entrez) %>%
    dplyr::filter(!is.na(`NCBI gene ID`)) %>%
    .[["NCBI gene ID"]] %>%
    unique()
```

```{R GO analysis}
goResults <- goana(de = de, universe = uv, species = "Hs")
head(goResults)
goResults %>% 
    rownames_to_column("id") %>%
    left_join(goSummaries) %>%
    as_tibble() %>%
    dplyr::filter(shortest_path> 3, DE > 1) %>%
    arrange(P.DE) %>%
    mutate(adjP = p.adjust(P.DE, "bonferroni")) %>%
    dplyr::filter(adjP < 0.05) %>%
    pander(caption = "GO Terms potentially enriched in the set of differentially expressed genes")
```

##improve GO analysis
##Q96K97del-specific transcript checking
##WGCNA 
##IRE enrichment

# Session Info

```{r}
pander(sessionInfo())
```

